### PR TITLE
[OC-4] added new wrappers for choreography

### PR DESCRIPTION
**Pull request by fengjun-yang1**
* added ROS2 wrappers for
1) uploading new animation files (i.e. dance moves in .cha format) 
2) fetching the list of animated moves on a robot
3) fetching the list of all choreographs on a robot 
* changed the implementation of upload_dance to upload the choreography file contents, instead of the file paths. This allows for uploading dance with the driver on a remote computer.

## Checklist

- [x] I have broken down the change into small, reviewable sizes.
- [x] I have added appropriate tests to CI and documentation to Notion.
- [x] I certify that my change is not related to Spot functionalities or that I have already completed [appropriate hardware testing](https://github.com/bdaiinstitute/bdai/blob/main/hil_testing.md).

### CODE REVIEW

**Comment by heuristicus** [comment]
Needs a black format but other than that looks fine to me.

File: spot_wrapper/spot_dance.py [hunk 1]
```diff
@@ -1,36 +1,77 @@
...
+from google.protobuf import text_format
 
 
 class SpotDance:
     def __init__(
         self,
         robot: Robot,
         choreography_client: ChoreographyClient,
-        is_licensed_for_choreography: bool,
+        logger
```
Comment (amessing-bdai) [inline spot_wrapper/spot_dance.py:23]: Add typing (RcutilsLogger I believe)

File: spot_wrapper/spot_dance.py [hunk 2]
```diff
@@ -1,36 +1,77 @@
...
+                return False, "Failed to convert animation file to protobuf message: {}".format(e)
+            try:
+                self._logger.info("Uploading the name {}".format(animation_name))
+                upload_response = self._choreography_client.upload_animated_move(animation_pb, animation_name)
+            except Exception as e:
+                error_msg = "Failed to upload animation: {}".format(e)
+                return False, error_msg
+        return True, "Success"
+    
+    def list_all_dances(self) -> tuple[bool, str, list[str]]:
```
Comment (amessing-bdai) [inline spot_wrapper/spot_dance.py:48]: Tuple and List to support 3.8

File: spot_wrapper/wrapper.py [hunk 3]
```diff
@@ -2508,8 +2507,29 @@ def undock(self, timeout=20):
...
     @try_claim
-    def execute_dance(self, filepath):
-        return self._spot_dance.execute_dance(filepath)
+    def execute_dance(self, data):
+        if self._is_licensed_for_choreography:
+            return self._spot_dance.execute_dance(data)
+        else:
+            return False, "Spot is not licensed for choreography"
+
+    def upload_animation(self, animation_name : str, animation_file_content : str):
```
Comment (amessing-bdai) [inline spot_wrapper/wrapper.py:2516]: return type

Comment (fengjun-yang1) [reply spot_wrapper/wrapper.py]: fixed these. Thanks for catching them!

**Review by davidwatkins-bdai — APPROVED** [review]
LGTM. Please run `black .` in the repository and push again.

**Review by davidwatkins-bdai — APPROVED** [review]
LGTM

**Pull request merged**